_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: post
+name: docker-meets-continuous-deployment
+title: Docker meets Continuous Deployment
+date: 2016-03-01 00:00:00 +00:00
+author: Robert Firek
+image:
+    src: /assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment.png
+    attribution:
+        text: Track icon by Freepik (CC 3.0 BY)
+        href: http://www.freepik.com
+tags:
+- Continuous Deployment
+- Docker
+- TeamCity
+--- 
+
+About one year ago I had a first contact with Docker. This new kid on the block promised to relieve our poor computers from installation of all tools, languages, dependencies and operating systems. Isolated run environments emerged on developers computers.
+
+A conservative approach of my ops teammates didn't change my joy of using Docker. Despite many people describing Docker as tool writen by developers for development our industry found new ways of using images and containers. Images of our applications and services became units of deployment for tools like Kubernetes, Docker Swarm or Marathon.
+
+But how this images appear in these tools?
+
+## Setting The Scene
+
+From developer's perspective any application is manifested by a code. There is long way before our code find a good place to execute on a production environment. I want to show how you this process can be easier with **Docker** and **Continuous Deployment** pipeline. To do that we need a small application. Our application is going to expose some HTTP API which we are going to call after the deployment. Let's assume that will are going to use **Gradle** to build the application and **TeamCity** for build automation.
```

Comment:
- From "a" developer's ... 
- ... by code "but" there is still a long way to go before this code finds its way to a production environment. 
- I want to show "you" how this process... 
- ... and "a" **Continuous Deployment" pipeline. 
- pipeline. "First we need a small application with an HTTP API that we can call after it is deployed. 
- Let's assume that "we are using" **Gradle**
- **TeamCity** "as a Continous Integration" server.